src/mavsdk/plugins/telemetry/telemetry_impl.cpp

```diff
@@ -1451,6 +1451,213 @@ void TelemetryImpl::process_distance_sensor(const mavlink_message_t& message)
     distance_sensor_struct.current_distance_m =
         static_cast<float>(distance_sensor_msg.current_distance) * 1e-2f; // cm to m
 
+    MavSensorOrientation orientation =
```

Comment:
Can we maybe put that whole big switch block in a separate function? Like `extractOrientation(distance_sensor_msg);`?